### DISCUSSION TITLE
Make game names in GameSwitcher show titles set in miyoogamelist

### DISCUSSION THREAD

**Discussion by MrWheyne** [OnionUI/Onion#1658] [Ideas]
Currently GameSwitcher snapshots show rom names instead of how they were setup in miyoogamelist. This is fine most of the time because you can easily change file names, but for arcade titles it is not possible as MAME doesn't work with renamed roms. Suggested change would be for all games to show their set miyoogamelist titles.

Example game: Street Fighter III: Third Strike
Current GameSwitcher label: sfiii3
Patched label: Street Fighter III: Third Strike

**Comment by Nighted** [comment]
BIOS/arcade_lists/arcade-rom-names.txt

If you have this arcade games should show up properly. If they aren't showing names properly go into the arcade games and press Y then refesh.

**Reply by MrWheyne** [reply]
Didn't work. File's there.

**Reply by Nighted** [reply]
Sigh... it works trust me.

https://www.mediafire.com/file/c4b40h7zikc6w7a/Arcade_Names_Game_Switcher.mp4

**Reply by Bukkithead** [reply]
I think I might have figured something out with this.

I also have the arcade-rom-names.txt list populated (I believe it generates when you run the 'Generate game list for short name roms' tool in Tweaks), however my system wasn't using the names in that file. 

I renamed the miyoogamelist.xml in \Roms\ARCADE and refreshed the rom list, and while it has overwritten the names with the less neat names in the arcade-rom-names.txt file, they are now showing properly in the GameSwitcher.

So it seems to be that if you are using a miyoogamelist.xml, in some cases like myself and OP it can cause the short name rom titles to appear in the GameSwitcher rather than the properly formatted ones. I'm going to keep playing around with those files and see if there's a way to have the miyoogamelist.xml file and it still format the names properly. For now at least I believe I can work around it.

**Reply by Bukkithead** [reply]
So there is a workaround for now, albeit somewhat manual and clunky. OP's suggestion to allow miyoogamelist.xml to also override the ROM names showed in GameSwitcher would be a welcome addition.

In \BIOS\arcade_lists there is a 'full-arcade-rom-name-list.txt' file. When you run the 'Generate game list for short name roms' tool in Tweaks, it uses the names in here to generate an 'arcade-rom-names.txt' which determines how your short-name ROMs are displayed in both the game browser and GameSwitcher.

Currently, miyoogamelist.xml will change the ROM names in the game browser but not GameSwitcher (the point of this request). If you remove/rename the miyoogamelist.xml file for that system, it will revert to the names used in those .txt files.

If you then want to change any of the names it uses, you can manually edit the 'full-arcade-rom-name-list.txt' file (I would suggest making a backup copy of this first) for any short-name ROMs you have, and run the tool again. You may then need to refresh the ROM list again, I can't remember if that was necessary or not. It took a little while for me to tidy them all but this has worked fine for me for now.

**Comment by Bukkithead** [comment]
Came to ask the same thing as well.  miyoogamelist.xml exists in \Roms\ARCADE which formats them just fine in the game browser but in the game switcher it still shows the ROM short names.

**Comment by Nighted** [comment]
miyoogamelist.xml, I don't use these. You can delete them.

**Reply by Bukkithead** [reply]
That's fine if you personally don't use them, but they are a feature of OnionOS (to include scraping metadata and properly formatted ROM titles), and the point of this request is to change it so that they do work properly with GameSwitcher, another feature of OnionOS. Just saying "don't use them" doesn't really make sense here.

**Reply by Nighted** [reply]
Sure it does. I used the usual dat files to meticulously and tediously hash and name my ROMs properly. Why would I need the miyoogamelist.xml files? Seems superfluous to me. Those files may also be giving people issues when it comes to getting proper arcade names in the game switcher. I don't use them, they can be deIeted without issue and I don't have issues with arcade names. I think it's quite relevant.

I'm not here to not make sense. I'm here to help solve this issue for those experiencing it.

**Reply by Bukkithead** [reply]
This isn't logged as an issue though, it's logged as a feature request. Just coming in and saying "well just don't use that feature then" is not helpful to anyone.

If you personally don't want to make use of those files, that's fine, they clearly have some drawbacks. However, they are a part of OnionOS and detailed several times in the FAQ, not just some random superfluous files. The whole point of the miyoogamelist.xml is to quickly and easily generate a list of properly formatted ROM names, and specifically avoid the process of 'meticulously and tediously hashing and naming ROMs properly', as it can just be generated automatically by something like Skraper.

It's a perfectly valid and reasonable request to update GameSwitcher so that it also works with those OnionOS miyoogamelist.xml files instead of having to do it the manual way. We're already aware that's the alternative.

**Reply by Nighted** [reply]
No. You don't need them and can delete them. I'm sure that it will be helpful to someone more so than contrarian pontification.

You seem very passionate about it and can obviously speak for everyone. Enjoy your crusade. 🙄
